test/OpenTelemetry.Exporter.Geneva.Tests/UnixUserEventsDataTransportTests.cs

```diff
@@ -143,6 +144,106 @@ public void UserEvents_Disabled_Success_Linux()
+    {
+        EnsureUserEventsEnabled();
+
+        var listener = new PerfTracepointListener(
+            "MicrosoftOpenTelemetryLogs_L4K1",
+            MetricUnixUserEventsDataTransport.MetricsTracepointNameArgs);
+
+        var logsTracepoint = UnixUserEventsDataTransport.Instance.RegisterUserEventProviderForLogs();
+
+        if (listener.IsEnabled())
+        {
+            throw new NotSupportedException($"{MetricUnixUserEventsDataTransport.MetricsTracepointName} is already enabled");
+        }
+
+        try
+        {
+            listener.Enable();
+
+            Console.WriteLine("------------- ready to write events -------------");
+            Thread.Sleep(5000);
+
+            if (logsTracepoint.IsEnabled)
+            {
+                var eventBuilder = CreateEventHeaderDynamicBuilder();
+
+                Console.WriteLine("About to write tracepoint:");
+                eventBuilder.Write(logsTracepoint);
+                Console.WriteLine("Written tracepoint.");
+            }
+
+            Thread.Sleep(5000);
+
+            this.testOutputHelper.WriteLine("Writing events from listener:");
+            Console.WriteLine("Writing events from listener:");
+            foreach (var e in listener.Events)
+            {
+                this.testOutputHelper.WriteLine(string.Join(", ", e.Select(kvp => $"{kvp.Key}={kvp.Value}")));
+                Console.WriteLine(string.Join(", ", e.Select(kvp => $"{kvp.Key}={kvp.Value}")));
+            }
+
+            this.testOutputHelper.WriteLine("Total events: " + listener.Events.Count);
+            Console.WriteLine("Total events: " + listener.Events.Count);
+        }
+        finally
+        {
+            try
+            {
+                listener.Disable();
+            }
+            catch
+            {
+            }
+
+            listener.Dispose();
+        }
+    }
+
+    private static EventHeaderDynamicBuilder CreateEventHeaderDynamicBuilder()
+    {
+        var eb = new EventHeaderDynamicBuilder();
+        eb.Reset("_");
+        eb.AddUInt16("__csver__", 1024, Microsoft.LinuxTracepoints.EventHeaderFieldFormat.HexInt);
+
+        eb.AddStructWithMetadataPosition("partA", out var partAFieldsCountMetadataPosition);
+
+        string rfc3339String = DateTime.UtcNow.ToUniversalTime().ToString("yyyy-MM-ddTHH:mm:ss.FFFFFFZ", CultureInfo.InvariantCulture);
+        eb.AddString16("time", rfc3339String);
+
+        byte partAFieldsCount = 1;
+
+        eb.SetStructFieldCount(partAFieldsCountMetadataPosition, partAFieldsCount);
+
+        // Part B
+
+        byte partBFieldsCount = 4; // We at least have three fields in Part B: _typeName, severityText, severityNumber, name
+        eb.AddStructWithMetadataPosition("PartB", out var partBFieldsCountMetadataPosition);
+        eb.AddString16("_typeName", "Log");
+        eb.AddUInt8("severityNumber", 21);
+
+        eb.AddString16("severityText", "Critical");
+        eb.AddString16("name", "CheckoutFailed");
+
```

Comment:
eventid and event name too?

Reply:
This was supposed to be a made-up event for testing data transport works, so I didn't make it a conforming event.